src/fieldsArray.js

```diff
@@ -6,98 +6,65 @@ const debug = require("debug")("passkit:fields");
-    } else {
-      this.fieldsKeys.add(key);
-    }
-  }
-
-  deleteFieldKey(key) {
-    this.fieldsKeys.delete(key);
-  }
-
-  emptyUnique() {
-    this.fieldsKeys.clear();
-  }
+const poolSymbol = Symbol("pool");
+
+class FieldsArray extends Array {
+	
+	constructor(pool,...args) {
+    	super(...args);
+    	this[poolSymbol] = pool;
+	}
+
+	/**
+	 * Like `Array.prototype.push` but will alter
+	 * also uniqueKeys set.
+	 */
+
+	push(...fieldsData) {
+		const validFields = fieldsData.reduce((acc, current) => {
+			if (!(typeof current === "object") || !schema.isValid(current, "field")) {
+				return acc;
+			}
+
+			if (acc.some(e => e.key === current.key) || this[poolSymbol].has(current.key)) {
+				debug(`Field with key "${key}" discarded: fields must be unique in pass scope.`);
+			} 
+
+			this[poolSymbol].add(current.key)
+			acc.push(current)
+
+			return acc;
+		}, []);
+
+		return Array.prototype.push.call(this, ...validFields);
+	}
+
+	/**
+	 * Like `Array.prototype.pop`, but will alter
+	 * also uniqueKeys set
+	 */
+
+	pop() {
+	 	const element = Array.prototype.pop.call(this);
+		this[poolSymbol].delete(element.key)
+    	return element;
+  	}
+
+	/**
+	 * Like `Array.prototype.splice` but will alter
+	 * also uniqueKeys set
+	 */
+
+	splice(start, deleteCount, ...items) {
+		const removeList = this.slice(start, deleteCount+start);
+		removeList.forEach(item => this[poolSymbol].delete(item.key));
+
+		return Array.prototype.splice.call(this, start, deleteCount, items);
+	}
+
+	get length() {
+		return this.length;
+	}
 }
 
-
-class ItemsArray extends Array {
-  constructor(owner) {
-    super();
-
-    this.owner = owner;
-  }
-
-  /**
-   * Like `Array.prototype.push` but will alter
-   * also uniqueKeys set.
-   */
-
-  push(...fieldsData) {
-    const validFields = fieldsData.reduce((acc, current) => {
-      if (!(typeof current === "object") || !schema.isValid(current, "field")) {
-        return acc;
-      }
-
-      if (acc.some(e => e.key === current.key)) {
-        debug(`Field with key "${key}" discarded: fields must be unique in pass scope.`);
-      } 
-
-      this.owner.addFieldKey(current.key)
-      acc.push(current)
-
-      return acc;
-    }, []);
-
-    return Array.prototype.push.call(this, ...validFields);
-  }
-
-  /**
-   * Like `Array.prototype.pop`, but will alter
-   * also uniqueKeys set
-   */
-
-  pop() {
-    const element = Array.prototype.pop.call(this);
-
-    this.owner.deleteFieldKey(element.key)
-    return element;
-  }
-
-  /**
-   * Like `Array.prototype.splice` but will alter
-   * also uniqueKeys set
-   */
-
-  splice(start, deleteCount, ...items) {
-    const removeList = this.slice(start, deleteCount+start);
-
-    removeList.forEach(item => this.owner.deleteFieldKey(item.key));
-
-    return Array.prototype.splice.call(this, start, deleteCount, items);
-  }
-
-  get length() {
-    return this.length;
-  }
-}
-
-
-module.exports = FieldsArray;
+module.exports = FieldsArray;
```

Comment:
@ianbale An empty end-line is always suggested to be inserted. Can you please insert it? Because if you won't do that, at my next edit or file opening, my editor will do that and I'll have to create a commit only for this or leave this as is (and causing a loop of auto-inserts, for me). At the end of the PR creation, I'll squash all your commits, so there won't be problems 😉